model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/organization/InfinispanOrganizationProvider.java

```diff
@@ -0,0 +1,222 @@
+        return getCacheDelegates(jpaOrgDelegate.getAllStream(attributes, first, max));
+    }
+
+    @Override
+    public boolean remove(OrganizationModel organization) {
+        registerOrganizationInvalidation(organization.getId());
+        return jpaOrgDelegate.remove(organization);
+    }
+
+    @Override
+    public void removeAll() {
+        //TODO: won't scale, requires a better mechanism for bulk deleting organizations within a realm
+        //this way, all organizations in the realm will be invalidated ... or should it be invalidated whole realm instead?
+        getAllStream().forEach(this::remove);
+    }
+
+    @Override
+    public boolean addMember(OrganizationModel organization, UserModel user) {
+        return jpaOrgDelegate.addMember(organization, user);
+    }
+
+    @Override
+    public boolean removeMember(OrganizationModel organization, UserModel member) {
+        return jpaOrgDelegate.removeMember(organization, member);
+    }
+
+    @Override
+    public Stream<UserModel> getMembersStream(OrganizationModel organization, String search, Boolean exact, Integer first, Integer max) {
+        return jpaOrgDelegate.getMembersStream(organization, search, exact, first, max);
+    }
+
+    @Override
+    public UserModel getMemberById(OrganizationModel organization, String id) {
+        return jpaOrgDelegate.getMemberById(organization, id);
+    }
+
+    @Override
+    public OrganizationModel getByMember(UserModel member) {
+        // Return cache delegate to ensure cache invalidation during write operations
+        return getCacheDelegate(jpaOrgDelegate.getByMember(member));
+    }
+
+    @Override
+    public boolean isManagedMember(OrganizationModel organization, UserModel member) {
+        return jpaOrgDelegate.isManagedMember(organization, member);
+    }
+
+    @Override
+    public boolean addIdentityProvider(OrganizationModel organization, IdentityProviderModel identityProvider) {
+        boolean added = jpaOrgDelegate.addIdentityProvider(organization, identityProvider);
+        if (added) {
+            registerOrganizationInvalidation(organization.getId());
+        }
+        return added;
+    }
+
+    @Override
+    public Stream<IdentityProviderModel> getIdentityProviders(OrganizationModel organization) {
+        return jpaOrgDelegate.getIdentityProviders(organization);
+    }
+
+    @Override
+    public boolean removeIdentityProvider(OrganizationModel organization, IdentityProviderModel identityProvider) {
+        boolean removed = jpaOrgDelegate.removeIdentityProvider(organization, identityProvider);
+        if (removed) {
+            registerOrganizationInvalidation(organization.getId());
+        }
+        return removed;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return getRealm().isOrganizationsEnabled();
+    }
+
+    @Override
+    public long count() {
+        return jpaOrgDelegate.count();
```

Comment:
This method is used in called from critical places like during authentication and token mapping. Can we have a cache here too?